bicep/main.bicep

```diff
@@ -171,21 +170,17 @@ var appGwSubnetId = ingressApplicationGateway ? (custom_vnet ? network.outputs.a
 
 @description('The full Azure resource ID of the DNS zone to use for the AKS cluster')
 param dnsZoneId string = ''
-var dnsZoneRg = !empty(dnsZoneId) ? split(dnsZoneId, '/')[4] : ''
-var dnsZoneName = !empty(dnsZoneId) ? split(dnsZoneId, '/')[8] : ''
-var isPrivate = !empty(dnsZoneId) ? split(dnsZoneId, '/')[7] == 'privateDnsZones' : false
+var isDnsZonePrivate = !empty(dnsZoneId) ? split(dnsZoneId, '/')[7] == 'privateDnsZones' : false
 
-module dnsZone './dnsZone.bicep' = if (!empty(dnsZoneId)) {
+module dnsZone './dnsZoneRbac.bicep' = if (!empty(dnsZoneId)) {
   name: 'addDnsContributor'
-  scope: resourceGroup(dnsZoneRg)
   params: {
-    dnsZoneName: dnsZoneName
-    isPrivate: isPrivate
```

Comment:
@Gordonby i think there are a bunch of issues here, you're not passing `dnsZoneId` or any parts of it to the module

Reply:
Lol. Good spot.
For some reason the module allowed that param to be optional.

All fixed up.